crates/rolldown/src/utils/pre_process_ecma_ast.rs

```diff
@@ -46,10 +50,21 @@ pub fn pre_process_ecma_ast(
     }
   }
 
-  ast.program.with_mut(|fields| {
+  ast.program.with_mut(|fields| -> anyhow::Result<()> {
+    let define_config =
+      ReplaceGlobalDefinesConfig::new(&bundle_options.define).map_err(|errs| {
+        anyhow::format_err!(
+          "Failed to generate defines config from {:?}. Got {:#?}",
+          bundle_options.define,
+          errs
+        )
+      })?;
+    ReplaceGlobalDefines::new(fields.allocator, define_config).build(fields.program);
```

Comment:
Construct the `ReplaceGlobalConfig` looks none trivial https://github.com/oxc-project/oxc/blob/910f2a0a0a7a4298923afc88511d089843d02d3a/crates/oxc_minifier/src/ast_passes/replace_global_defines.rs#L30-L54, do we need to share them with `Arc`?

Reply:
Good catch. @Boshen cc

Reply:
Guess this also requires https://github.com/oxc-project/oxc/issues/4284.

Reply:
https://github.com/oxc-project/oxc/blob/69dc6c9cb1f93209e5a0bc003918d1bc6f50e455/crates/oxc_minifier/src/plugins/replace_global_defines.rs#L17 it's `Arc`ed.

Reply:
Yeah, the `Defiens` is `Arc`ed in oxc side,but we still create the `DefineConfig` per file.